File: src/main/java/com/widehouse/dnd5e/equipment/Armor.java
package com.widehouse.dnd5e.equipment;

import lombok.Getter;

@Getter
public class Armor extends Item {
    public enum ArmorType {
        LIGHT_ARMOR, MEDIUM_ARMOR, HEAVY_ARMOR, SHIELD
    }

    private ArmorType armorType;

    public Armor(String name, Integer cost, Integer weight, ArmorType armorType) {
        super(name, cost, weight);
        this.armorType = armorType;
    }
}

File: src/main/kotlin/com/widehouse/dnd/character/item/Armor.kt
package com.widehouse.dnd.character.item

class Armor(
    override val name: String,
    val itemType: ArmorType,
    val armorClass: Int,
    override val cost: Coin = Coin(0),
    override val weight: Int = 0
) : Item(name, cost, weight)

enum class ArmorType {
    LightArmor,
    MediumArmor,
    HeavyArmor
}
